src/main/java/bio/overture/songsearch/graphql/AnalysisDataFetcher.java

```diff
@@ -56,4 +68,74 @@ public AnalysisDataFetcher(AnalysisService analysisService) {
       return analysisService.getAnalyses(filter.build(), page.build());
     };
   }
+
+  public DataFetcher<List<SampleMatchedAnalysisPair>> getSampleMatchedAnalysisPairsFetcher() {
+    return env -> {
+      val analysisId = env.getArguments().get("analysisId").toString();
+
+      val analysisFromId = analysisService.getAnalysisById(analysisId);
+      val samples = getFlattenedSamplesFromAnalysis(analysisFromId);
+
+      validateAnalysisAndSamplesValidForQuery(analysisFromId, samples);
+
+      val flattenedSampleOfInterest = (FlatDonorSample) samples.get(0);
```

Comment:
why the cast?

Reply:
what the... not sure where that came from, removed :+1: